larpandoracontent/LArHelpers/LArMCParticleHelper.cc

```diff
@@ -57,6 +57,18 @@ bool LArMCParticleHelper::IsBeamParticle(const MCParticle *const pMCParticle)
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+bool LArMCParticleHelper::IsLeadingBeamParticle(const MCParticle *const pMCParticle)
+{
+    // ATTN: Only the parent triggered beam particle has nuance code 2001
+    const int parentNuance(LArMCParticleHelper::GetNuanceCode(LArMCParticleHelper::GetParentMCParticle(pMCParticle)));
+
+    bool isLeading(LArMCParticleHelper::IsLeading(pMCParticle) && (parentNuance == 2001 || parentNuance == 2000));
```

Comment:
Need to review where the actual values of the nuance codes are used, and ensure that it's in the absolute minimum number of places: ideally just in one or two helper functions, so that the values don't appear anywhere else. (const bool)